### PR TITLE
Add support for macOS to SysInfo

### PR DESCRIPTION
**Pull request by q-p**
(tested to work on 10.14 Mojave, but should work on older versions, too)

### CODE REVIEW

**Comment by ax3l** [comment]
Dear @q-p, thank you very much for the contribution, that's wonderful!
Related to #492

Did you encounter any other issues on OSX? We still have to add CI for it to support it permanently.

**Review by ax3l — DISMISSED** [review]

**Comment by q-p** [comment]
I'm currently working on integrating a DLR in-house library (SPLiSS) using alpaka, and all I can say for now is that with these changes it seems to compile (and that the modified SysInfo methods return the correct values by calling them manually). :)

**Review by psychocoderHPC — DISMISSED** [review]

File: include/alpaka/dev/cpu/SysInfo.hpp [hunk 1]
```diff
@@ -228,7 +228,19 @@ namespace alpaka
                         throw std::runtime_error("Unable to open '/proc/meminfo'!");
                     }
 #elif BOOST_OS_MACOS
-    #error "getFreeGlobalMemSizeBytes not implemented for __APPLE__!"
+                    int free_pages = 0;
+                    std::size_t len = sizeof(free_pages);
+                    if(sysctlbyname("vm.page_free_count", &free_pages, &len, NULL, 0) < 0)
+                    {
+                      throw std::logic_error("getFreeGlobalMemSizeBytes failed calling sysctl(vm.page_free_count)!");
```
Comment (BenjaminW3) [inline include/alpaka/dev/cpu/SysInfo.hpp]: Could you please use 4 spaces indentation?

File: include/alpaka/dev/cpu/SysInfo.hpp [hunk 2]
```diff
@@ -228,7 +228,19 @@ namespace alpaka
...
+                    }
+                    int page_size = 0;
+                    len = sizeof(page_size);
+                    if(sysctlbyname("vm.pagesize", &page_size, &len, NULL, 0) < 0)
+                    {
+                      throw std::logic_error("getFreeGlobalMemSizeBytes failed calling sysctl(vm.pagesize)!");
```
Comment (BenjaminW3) [inline include/alpaka/dev/cpu/SysInfo.hpp]: here as well

**Review by BenjaminW3 — DISMISSED** [review]
Thank you for your help!

**Review by ax3l — APPROVED** [review]

**Pull request merged**